### PR TITLE
Add progress bar

### PR DESCRIPTION
**Pull request by wolfsolver**
Add progress bar in message info.

This is done by adding 2 local var to class ZCL_ALOG_PROGRESS_REP_LOGGER and 2 relative method and by changing ENTRY_INTERNAL with effective progress.
New Variable:
variable
New Method:
method

Result:
Progress

### CODE REVIEW

**Comment by flaiker** [comment]
Hi, thanks. I'll review the PR later today.

File: src/example/zalog_example_prog_rep_logger.prog.abap [hunk 1]
```diff
@@ -3,7 +3,12 @@ REPORT zalog_example_prog_rep_logger.
 INITIALIZATION.
   DATA(go_logger) = NEW zcl_alog_progress_rep_logger( ).
 
+* EP Set Total Entries
+  go_logger->set_TOTAL_ENTRIES( 10 ).
```
Comment (flaiker) [inline src/example/zalog_example_prog_rep_logger.prog.abap]: What does `EP` stand for? The comment also seems really unnecessary.

Also please pretty print anything non-statement to lower-case.

File: src/example/zalog_example_prog_rep_logger.prog.abap [hunk 2]
```diff
@@ -3,7 +3,12 @@ REPORT zalog_example_prog_rep_logger.
...
+
   DO 10 TIMES.
+*   EP Set Actual Entries
+    go_logger->set_actual_ENTRIES( sy-index ).
```
Comment (flaiker) [inline src/example/zalog_example_prog_rep_logger.prog.abap]: Same here

File: src/zcl_alog_progress_rep_logger.clas.abap [hunk 3]
```diff
@@ -12,36 +12,59 @@
...
 
+
   METHOD entry_internal.
     cl_progress_indicator=>progress_indicate( i_text               = iv_text
                                               i_output_immediately = mv_output_immediately
-                                              i_processed          = 1
-                                              i_total              = 100 ).
+* EP use maximun message
+*                                              i_processed          = 1
+*                                              i_total              = 100 ).
```
Comment (flaiker) [inline src/zcl_alog_progress_rep_logger.clas.abap]: Why is this commented out? Just delete it.

File: src/zcl_alog_progress_rep_logger.clas.abap [hunk 4]
```diff
@@ -53,4 +76,14 @@ CLASS zcl_alog_progress_rep_logger IMPLEMENTATION.
                                               i_processed          = 1
                                               i_total              = 100 ).
```
Comment (flaiker) [inline src/zcl_alog_progress_rep_logger.clas.abap]: These should also use `mv_actual_entries` and `mv_total_entries`, shouldn't they?

**Review by fabianlupa — CHANGES_REQUESTED** [review]
Just some minor things.

The previous behavior seems really suboptimal especially considering in Belize _only_ the percentage is displayed and no watch animation:
image

File: src/example/zalog_example_prog_rep_logger.prog.abap (hunk 1 above)
Comment (wolfsolver) [reply src/example/zalog_example_prog_rep_logger.prog.abap]: fixed. just for your info, i always sign my code with "EP" it's my initial letter :-)

File: src/zcl_alog_progress_rep_logger.clas.abap (hunk 4 above)
Comment (wolfsolver) [reply src/zcl_alog_progress_rep_logger.clas.abap]: yes, you are right, there are two point and I change only one.

**Review by fabianlupa — APPROVED** [review]

**Comment by fabianlupa** [comment]
Thanks!

**Pull request merged**